src/lib/hofs/functions.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * This method creates a function that invokes func once it’s called n or more times.
+ * @see https://youmightnotneed.com/lodash#after
+ * @template A
+ * @template R
+ * @param {number} times
+ * @param {(...args: A[]) => R} fn
+ * @returns {(...args: A[]) => void | R}
+ */
+export const after = (times, fn) => {
+  let counter = 0
+  /**
+   * @type {(...args: A[]) => void | R}
+   */
+  return (...args) => {
+    counter++
+    if (counter >= times) {
+      return fn(...args)
+    }
+  }
+}
+
+/**
+ * @see https://youmightnotneed.com/lodash#once
+ * @template A
+ * @template R
+ * @param {(...args: A[]) => R} fn
+ * @returns {(...args: A[]) => R}
+ */
+export const once = (fn) => {
+  let called = false
+  /**
+   * @type {R}
+   */
+  let result
+
+  /**
+   * @type {(...args: A[]) => R}
+   */
+  return (...args) => {
+    if (!called) {
+      result = fn(...args)
+      called = true
+    }
+    return result
+  }
+}
+
+/**
+ * @see https://youmightnotneed.com/lodash#debounce
+ *
+ * @template A
+ * @template R
+ * @param {(...args: A[]) => R} fn - The function to debounce.
+ * @param {number} delay - The number of milliseconds to delay.
+ * @param {Object} options
+ * @param {boolean} [options.leading]
+ * @returns {(...args: A[]) => void}
+ */
+export const debounce = (fn, delay, { leading = false } = {}) => {
```

Comment:
other params are not because higher order functions are composed of composable functions, which in general require  a certain format `(fn, ...args)`

I changed the functions in this file to use this format.